napi/src/transformer.rs

```diff
@@ -755,9 +755,8 @@ impl<'de, V: serde::Deserialize<'de>, const IS_VEC: bool> serde::Deserialize<'de
     D: serde::Deserializer<'de>,
   {
     use serde::Deserializer;
-    let content = serde::__private::de::Content::deserialize(deserializer)?;
-    let de: serde::__private::de::ContentRefDeserializer<D::Error> =
-      serde::__private::de::ContentRefDeserializer::new(&content);
+    let content = serde_content::Value::deserialize(deserializer)?;
+    let de = serde_content::Deserializer::new(content.clone()).coerce_numbers();
```

Comment:
This clone is theoretically cheap if the content contains borrowed data since `serde_content::Value` uses `Cow` internally.